hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyPurgeRequest.java

```diff
@@ -64,10 +74,27 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       }
     }
 
-    omClientResponse = new OMKeyPurgeResponse(omResponse.build(),
-          keysToBePurgedList);
-    addResponseToDoubleBuffer(trxnLogIndex, omClientResponse,
-        omDoubleBufferHelper);
+    try {
+      OmSnapshot omFromSnapshot = null;
+      if (fromSnapshot != null) {
+        SnapshotInfo snapshotInfo =
+            ozoneManager.getMetadataManager().getSnapshotInfoTable()
+                .get(fromSnapshot);
+        omFromSnapshot = (OmSnapshot) omSnapshotManager
+            .checkForSnapshot(snapshotInfo.getVolumeName(),
+                snapshotInfo.getBucketName(),
+                getSnapshotPrefix(snapshotInfo.getName()));
+      }
+
```

Comment:
Shall we move this "check and get" to `OMKeyPurgeResponse` instead?

i.e. What could go wrong if we just pass `fromSnapshot` to `OMKeyPurgeResponse` constructor (rather than `omFromSnapshot`)?

Here in `validateAndUpdateCache` we could just sanity check the snapshot existence.

Reply:
The problem is `OmSnapshotManager` is not part of `OMMetadataManager` and it is part of `OzoneManager` so we won't be able to get `OmSnapshot` in `OMKeyPurgeResponse` as it only has `OMMetadataManager`

Reply:
Ok then. Let's leave this part of the logic here.